File: app/src/main/java/com/kakeibo/ui/categories/CategoryGridAdapter.java
package com.kakeibo.ui.categories;

import android.view.LayoutInflater;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.databinding.DataBindingUtil;
import androidx.recyclerview.widget.RecyclerView;

import com.kakeibo.R;
import com.kakeibo.data.CategoryStatus;
import com.kakeibo.databinding.ItemGridBinding;
import com.kakeibo.ui.ItemSaveListener;

import java.util.ArrayList;
import java.util.List;

/*
 * Used in TabFragment1, Settings
 */
public class CategoryGridAdapter extends RecyclerView.Adapter<CategoryGridAdapter.ViewHolder> {
    private static final String TAG = CategoryGridAdapter.class.getSimpleName();

    private final ItemSaveListener _itemSaveListener;

    private List<CategoryStatus> _categoryStatusList = new ArrayList<>();

    public CategoryGridAdapter(ItemSaveListener itemSaveListener) {
        this._itemSaveListener = itemSaveListener;
    }

    public void setCategoryStatuses(List<CategoryStatus> categoryStatusList) {
        this._categoryStatusList = categoryStatusList;
        notifyDataSetChanged();
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        ItemGridBinding itemGridBinding = DataBindingUtil.inflate(
                LayoutInflater.from(parent.getContext()), R.layout.item_grid, parent, false);
        return new ViewHolder(itemGridBinding);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        CategoryStatus categoryStatus = _categoryStatusList.get(position);
        holder.bind(categoryStatus, _itemSaveListener);
    }

    @Override
    public int getItemCount() {
        return _categoryStatusList != null? _categoryStatusList.size(): 0;
    }

    static class ViewHolder extends RecyclerView.ViewHolder {
        final private ItemGridBinding itemGridBinding;

         ViewHolder (ItemGridBinding itemGridBinding) {
            super(itemGridBinding.getRoot());
            this.itemGridBinding = itemGridBinding;
        }

        public void bind(CategoryStatus categoryStatus, ItemSaveListener itemSaveListener) {
            itemGridBinding.setCategory(categoryStatus);
            itemGridBinding.setItemSaveListener(itemSaveListener);
            itemGridBinding.executePendingBindings();
        }
    }
}

File: app/src/main/java/com/kakeibo/ui/categories/CategoryGridAdapter.kt
package com.kakeibo.ui.categories

import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.databinding.DataBindingUtil
import androidx.recyclerview.widget.RecyclerView
import com.kakeibo.R
import com.kakeibo.data.CategoryStatus
import com.kakeibo.databinding.ItemGridBinding
import com.kakeibo.ui.ItemSaveListener
import java.util.*

/*
* Used in TabFragment1, Settings
*/
class CategoryGridAdapter(
        private val _itemSaveListener: ItemSaveListener
        ) : RecyclerView.Adapter<CategoryGridAdapter.ViewHolder>() {

    private var _categoryStatusList: List<CategoryStatus>? = ArrayList()

    fun setCategoryStatuses(categoryStatusList: List<CategoryStatus>?) {
        _categoryStatusList = categoryStatusList
        notifyDataSetChanged()
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val itemGridBinding: ItemGridBinding = DataBindingUtil.inflate(
                LayoutInflater.from(parent.context),
                R.layout.item_grid, parent, false)
        return ViewHolder(itemGridBinding)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val categoryStatus = _categoryStatusList!![position]
        holder.bind(categoryStatus, _itemSaveListener)
    }

    override fun getItemCount(): Int {
        return if (_categoryStatusList != null) _categoryStatusList!!.size else 0
    }

    class ViewHolder(
            private val itemGridBinding: ItemGridBinding
            ) : RecyclerView.ViewHolder(itemGridBinding.root) {

        fun bind(categoryStatus: CategoryStatus?, itemSaveListener: ItemSaveListener?) {
            itemGridBinding.category = categoryStatus
            itemGridBinding.itemSaveListener = itemSaveListener
            itemGridBinding.executePendingBindings()
        }
    }
}